xblock/field_data.py

```diff
@@ -154,8 +156,24 @@ def _field_data(self, block, name):
 
         return self._scope_mappings[scope]
 
-    def get(self, block, name):
-        return self._field_data(block, name).get(block, name)
+    def _shared_field_data(self, remote_field_scope):
+        """Summary
+        
+        Args:
+            block (TYPE): Description
+            name (TYPE): Description
+        
+        Returns:
+            TYPE: Description
+        """
+        return self._scope_mappings[remote_field_scope]
+
+
+    def get(self, block, name, remote_field_scope=None):
+        if isinstance(block, XBlock):
+            return self._field_data(block, name).get(block, name)
+        elif isinstance(block, ScopeIds):
+            return self._shared_field_data(remote_field_scope).get(block, name)
```

Comment:
I'm a bit confused. This looks like it's trying to make `get` work with either an `XBlock` or the `ScopeIds` from an `XBlock`, but then the behavior is different in the different cases (rather than just pulling the `ScopeIds` out of the `XBlock` in the former case). Why?